core/iwasm/aot/aot_runtime.c

```diff
@@ -1476,7 +1478,16 @@ create_exports(AOTModuleInstance *module_inst, AOTModule *module,
     }
 
 #if WASM_ENABLE_MULTI_MEMORY == 0
-    bh_assert(module_inst->export_memory_count <= 1);
+    if (module_inst->export_memory_count) {
+        /* There may be multiple export memories but they must
+           point to the first memory */
+        bh_assert(module_inst->memory_count == 1);
+        for (i = 0; i < module->export_count; i++) {
+            if (exports[i].kind == EXPORT_KIND_MEMORY) {
+                bh_assert(exports[i].index == 0);
+            }
+        }
+    }
```

Comment:
It seems unnecessary to include the new assertion. Or, may add it at line 1473?

Reply:
Adding bh_assert at L1473 seems not so good since it only applies change for memory. I updated aot_loader.c to add checks for the exports' name, kind and index, just like what the wasm loader does, please help review again.